New steps added in the release step (#25)
10up/.github: CONTRIBUTING.md

2. Version bump: Bump the version number in `plugin.php`, `readme.txt`, and any other relevant files if it does not already reflect the version being released.
3. Changelog: Add/update the changelog in `CHANGELOG.md` and `readme.txt`.
4. Props: update `CREDITS.md` file with any new contributors, and confirm maintainers are accurate.
5. New files: Check to be sure any new files/paths that are unnecessary in the production version are included in `.gitattributes` or `.distignore`.
6. Readme updates: Make any other readme changes as necessary. `CHANGELOG.md` and `README.md` are geared toward GitHub and `readme.txt` contains WordPress.org-specific content. The two are slightly different.
7. Merge: Make a non-fast-forward merge from your release branch to `develop` (or merge the pull request), then do the same for `develop` into `trunk`, ensuring you pull the most recent changes into `develop` first (`git checkout develop && git pull origin develop && git checkout trunk && git merge --no-ff develop`). `trunk` contains the stable development version.
8. Push: Push your `trunk` branch to GitHub (e.g. `git push origin trunk`).

> 9. Compare develop and trunk to ensure no additional changes were missed. Visit [REPOSITORY_URL]/compare/trunk...develop

Reviewer: ```suggestion
9. Compare `trunk` to `develop` to ensure no additional changes were missed. Visit [REPOSITORY_URL]/compare/trunk...develop
```

> 9. Push: Push your `trunk` branch to GitHub (e.g. `git push origin trunk`).

Reviewer 2: I actually have another step after this that I tend to check myself (as I've run into issues in the past here) where I compare develop>trunk to ensure I didn't miss any changes.  So could add:
> 1. [Compare develop and trunk](https://github.com/10up/simple-local-avatars/compare/trunk...develop) to ensure no additional changes were missed.

Reviewer 2: Noting the link here is an example from SLA

Author: https://github.com/10up/.github/pull/25/commits/92a5c355286877c523823a610c80f933d89135ac

Added this:

_10. Compare develop and trunk to ensure no additional changes were missed. Visit [REPOSITORY_URL]/compare/trunk...develop_

Please let me know if any changes are required.

> 10. Test the pre-release ZIP locally by downloading it from the **Build release zip** action artifact to ensure the plugin doesn't break after release.

Reviewer: ```suggestion
10. Test the pre-release ZIP locally by downloading it from the **Build release zip** action artifact and installing it locally. Ensure this zip has all the files we expect, that it installs and activates correctly and that all basic functionality is working.
```

1. Branch: Starting from `develop`, cut a release branch named `release/X.Y.Z` for your changes.
2. Version bump: Bump the version number in `plugin.php`, `readme.txt`, and any other relevant files if it does not already reflect the version being released.
3. Changelog: Add/update the changelog in `CHANGELOG.md` and `readme.txt`.
4. Props: update `CREDITS.md` file with any new contributors, and confirm maintainers are accurate.
5. New files: Check to be sure any new files/paths that are unnecessary in the production version are included in `.gitattributes` or `.distignore`.
6. Readme updates: Make any other readme changes as necessary. `CHANGELOG.md` and `README.md` are geared toward GitHub and `readme.txt` contains WordPress.org-specific content. The two are slightly different.
7. Before proceeding to the next step, ensure you pull the latest changes from the `develop` branch locally. This will bring in the changes from the merged release PR.

> 8. Merge: Make a non-fast-forward merge from your release branch to `develop` (or merge the pull request), then do the same for `develop` into `trunk` (`git checkout trunk && git merge --no-ff develop`). `trunk` contains the stable development version.

Reviewer: So thinking about this again, I believe the order of these steps is wrong. There's nothing to pull into `develop` until after the release PR is merged into `develop`. I'd suggest removing the newly added step 7 and making the following change to this step:

```suggestion
8. Merge: Make a non-fast-forward merge from your release branch to `develop` (or merge the pull request), then do the same for `develop` into `trunk`, ensuring you pull the most recent changes into `develop` first (`git checkout develop && git pull origin develop && git checkout trunk && git merge --no-ff develop`). `trunk` contains the stable development version.
```
